src/odb/src/db/dbPGpin.cpp

```diff
@@ -0,0 +1,222 @@
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// Generator Code Begin Cpp
+#include <iostream>
+#include "dbPGpin.h"
+#include "db.h"
+#include "dbDatabase.h"
+#include "dbDiff.hpp"
+#include "dbTable.h"
+#include "dbTable.hpp"
+#include "dbTypes.h"
+// User Code Begin Includes
+#include <algorithm>
+
+#include "dbBPin.h"
+#include "dbBTerm.h"
+#include "dbBlock.h"
+#include "dbITerm.h"
+#include "dbLib.h"
+#include "dbMPin.h"
+#include "dbMaster.h"
+#include "dbTech.h"
+#include "dbTechLayer.h"
+#include "dbTechVia.h"
+#include "dbVia.h"
+#include "utl/Logger.h"
+#include <vector>
+#include <string>
+
+// User Code End Includes
+namespace odb {
+
+_dbPGpin::_dbPGpin(_dbDatabase* db)
+{
+  // User Code Begin Constructor
+  // User Code End Constructor
+}
+
+_dbPGpin::~_dbPGpin()
+{
+  // User Code Begin Destructor
+  // User Code End Destructor
+}
+
+int r2grid (int number, dbTech *db_tech) {
+    // This will round "number" to a value that respects the manufacturing grid
+    // For example, if the manufacturing grid is 5 microns and number = 20333,
+    // the procedure will output 20335.
+    auto grid = db_tech->getManufacturingGrid();
+    return round(number / double(grid) * grid);
+}
+
+void dbPGpin::create_PGpin (dbBlock* block, dbTech* tech, const char * source_net_name, int num_connection_points, const std::string& position) {
+  /*
+  _dbPGpin* obj = (_dbPGpin*) this;
+  dbDatabase* db = (dbDatabase*)obj->getDatabase();
+  
+  db->getTech();
+  dbBlock* block = db->getChip()->getBlock();
+  dbTech* tech = db->getTech();
+  */
+
+  dbNet* create_pin_net = block->findNet(source_net_name);
+  dbTechLayer* PGpin_metal_layer = tech->findLayer("met1"); 
+  dbSet<dbSWire> swires = create_pin_net->getSWires();
+  dbSBox* pdn_wire = new dbSBox();
+
+  int num = 0;
```

Comment:
This appears to be a boolean - perhaps bool first = true?

Reply:
changed